src/pages/SignIn.tsx

```diff
@@ -0,0 +1,61 @@
+import { useForm, SubmitHandler } from 'react-hook-form'
+import './../css/form.css'
+import { Link } from 'react-router-dom'
+
+type Inputs = {
+  name: string
+  email: string
+  password: string
+  emailRequired: string
+}
+
+type SignUpProps = {
+  signIn: (input: User) => void
+}
+
+function SignIn({ signIn }: SignUpProps) {
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<Inputs>()
+  const onSubmit: SubmitHandler<Inputs> = (data) => signIn(data)
+
+  watch('email')
+
+  return (
+    <div className="form-container">
+      <h2>Sign in</h2>
+      <form className="form" onSubmit={handleSubmit(onSubmit)}>
+        <div>
+          <label>email:</label>
+          <input
+            defaultValue="enter your email"
+            type="email"
+            {...register('email', { required: true })}
+          />
+          {errors.email && <span>This field is required</span>}
+        </div>
+        <div>
+          <label>password:</label>
+          <input
+            type="password"
+            {...register('password', { required: true })}
+          />
+          {errors.password && <span>This field is required</span>}
```

Comment:
I kinda love & hate RHF because it supports so many patterns. However for a field like password you commonly need a bunch of different validation requirements. On ADDI we have a standard where you always define the error message in the validation like so:
```suggestion
            {...register('password', { required: 'This field is required',  })}
          />
          {errors.password && <span>{errors.password.message}</span>}
```
Go ahead & add a `min` requirement to this field using RHF with its own error message